R/utilities-dimensions.R

```diff
@@ -0,0 +1,162 @@
+#' Dimension existence
+#'
+#' @param dimension String name of the dimension.
+#' @details Check if the provided dimension is supported.
+#' @export
+existsDimension <- function(dimension) {
+  enumHasKey(dimension, enum = Dimensions)
+}
+
+#' Validate dimension
+#'
+#' @param dimension String name of the dimension.
+#' @details Check if the provided dimension is supported. If not, throw an error
+#' @export
+validateDimension <- function(dimension) {
+  if (!existsDimension(dimension)) {
+    stop(messages$errorDimensionNotSupported(dimension))
+  }
+}
+
+#' Unit existence
+#'
+#' @param unit String name of the unit
+#' @param dimension String name of the dimension.
+#' @details Check if the unit is valid for the dimension.
+#' @export
+hasUnit <- function(unit, dimension) {
+  validateIsString(c(unit, dimension))
+  validateDimension(dimension)
+  dimensionTask <- getDimensionTask()
+  # Get the dimension object
+  dimension <- rClr::clrCall(dimensionTask, "DimensionByName", enc2utf8(dimension))
+  rClr::clrCall(dimension, "HasUnit", enc2utf8(unit))
+}
+
+#' Validate unit
+#'
+#' @param unit String name of the unit
+#' @param dimension String name of the dimension.
+#' @details Check if the unit is valid for the dimension. If not, throw an error
+#' @export
+validateUnit <- function(unit, dimension) {
+  if (!hasUnit(unit, dimension)) {
+    stop(messages$errorUnitNotSupported(unit, dimension))
+  }
+}
+
+#' Get base unit of a dimension
+#'
+#' @param dimension Dimension for which the base unit is returned.
+#'
+#' @return String name of the base unit.
+#' @export
+getBaseUnit <- function(dimension) {
+  validateDimension(dimension)
+  dimensionTask <- getDimensionTask()
+  # Get the dimension object
+  dimension <- rClr::clrCall(dimensionTask, "DimensionByName", enc2utf8(dimension))
+  unit <- rClr::clrCall(dimension, "get_BaseUnit")
+  rClr::clrCall(unit, "ToString")
+}
+
+#' Calculate factor for converting values of one dimension into another
+#'
+#' @param dimension1 String name of the first dimension
+#' @param dimension2 String name fo the second dimension
+#' @param mw Double value of the molecular weight of the molecule. Used
+#' for conversion between molar and mass concentrations or amounts. Can be \code{NULL}
+#' for other dimensions (default)
+#'
+#' @return A numerical factor the values of the first dimensions must be multiplied by to
+#' be converted to second dimension. The conversion assumes values being in base units
+#' of both dimensions.
+#' If the values cannot be transformed between the dimensions, an error is thrown.
+#' @export
+dimensionsConversionFactor <- function(dimension1, dimension2, mw = NULL) {
```

Comment:
This is very useful if you want e.g. to plot your data in molar or mass concentrations. Is there a better way to check if the dimensions can be converted than to explicitly consider every possible use case?